src/common/hook/usePublishQueue/usePublishQueue.test.tsx

```diff
@@ -59,10 +68,35 @@ describe("usePublishQueue", () => {
     mockGetPublishingJobs.mockReturnValueOnce(sampleJobs);
     mockPublishJob.mockResolvedValue("tx-id");
     mockPublishJob.mockRejectedValueOnce(new Error("Some error"));
+    mockUploadToStorage.mockReturnValue(uploadSuccess);
     const { result } = renderHook(() => usePublishQueue(config, formEntires));
     await act(async () => {
       await result.current.publish();
     });
     expect(result.current.failedPublishedDocuments).toHaveLength(1);
   });
+
+  it("should publish correctly if uploaded document is working", async () => {
+    mockGetPublishingJobs.mockReturnValueOnce(sampleJobs);
+    mockPublishJob.mockResolvedValue("tx-id");
+    mockUploadToStorage.mockReturnValue(uploadSuccess);
+
+    const { result } = renderHook(() => usePublishQueue(config, formEntires));
+    await act(async () => {
+      await result.current.publish();
+    });
+
+    expect(result.current.publishedDocuments).toHaveLength(3);
+  });
+
+  it("should have error if getPublishJobs throw error", async () => {
+    mockGetPublishingJobs.mockRejectedValue(new Error("some Error"));
+    mockPublishJob.mockResolvedValue("tx-id");
+    const { result } = renderHook(() => usePublishQueue(config, formEntires));
+    await act(async () => {
+      await result.current.publish();
+    });
+
+    expect(result.current.error).toStrictEqual(new Error("some Error"));
```

Comment:
should it assert failed jobs too?

Reply:
at this point, should not because it failed to get the publishing jobs. the error message will be displayed on the error page and be able to download in the error logs.